### ISSUE TITLE
Crash on Startup - Possible sodium conflict?

### ISSUE THREAD

**Issue by MaxJW** [RaphiMC/ImmediatelyFast#109]
Mod Version: 1.2.0+1.20.1
Description: Updated to the latest version, now crashes on startup so had to remove. Possible an incompatibility with sodium as that appears to be the error crash here? I can provide a mod list as well if you think it is another mod incompatbility though!

<details>

<summary>Error Log</summary>

```
A mod crashed on startup!
net.fabricmc.loader.impl.FormattedException: java.lang.RuntimeException: Could not execute entrypoint stage 'preLaunch' due to errors, provided by 'sodium'!
	at net.fabricmc.loader.impl.FormattedException.ofLocalized(FormattedException.java:63)
	at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:164)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:68)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
Caused by: java.lang.RuntimeException: Could not execute entrypoint stage 'preLaunch' due to errors, provided by 'sodium'!
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.lambda$invoke0$0(EntrypointUtils.java:51)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:49)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35)
	at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:162)
	... 2 more
	Suppressed: java.lang.ExceptionInInitializerError
		at java.base/java.lang.Class.forName0(Native Method)
		at java.base/java.lang.Class.forName(Class.java:467)
		at net.fabricmc.loader.impl.util.DefaultLanguageAdapter.create(DefaultLanguageAdapter.java:50)
		at net.fabricmc.loader.impl.entrypoint.EntrypointStorage$NewEntry.getOrCreate(EntrypointStorage.java:117)
		at net.fabricmc.loader.impl.entrypoint.EntrypointContainerImpl.getEntrypoint(EntrypointContainerImpl.java:53)
		at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:47)
		... 4 more
	Caused by: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_2960 failed
		at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:427)
		at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323)
		at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218)
		at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112)
		at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
		at dev.stashy.soundcategories.SoundCategories.<clinit>(SoundCategories.java:41)
		... 10 more
	Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
		at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392)
		at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234)
		at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202)
		at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422)
		... 15 more
	Caused by: java.lang.ClassCastException: class com.llamalad7.mixinextras.lib.apache.commons.tuple.ImmutablePair cannot be cast to class org.apache.commons.lang3.tuple.Pair (com.llamalad7.mixinextras.lib.apache.commons.tuple.ImmutablePair and org.apache.commons.lang3.tuple.Pair are in unnamed module of loader net.fabricmc.loader.impl.launch.knot.KnotClassLoader @38364841)
		at ca.fxco.memoryleakfix.config.mixinExtension.UnMixinExtension.preApply(UnMixinExtension.java:23)
		at org.spongepowered.asm.mixin.transformer.ext.Extensions.preApply(Extensions.java:156)
		at org.spongepowered.asm.mixin.transformer.TargetClassContext.preApply(TargetClassContext.java:413)
		at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:401)
		at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
		... 18 more
Caused by: java.lang.ExceptionInInitializerError
	at me.jellysquid.mods.sodium.client.util.workarounds.probe.GraphicsAdapterProbe.findAdapters(GraphicsAdapterProbe.java:26)
	at me.jellysquid.mods.sodium.client.SodiumPreLaunch.onPreLaunch(SodiumPreLaunch.java:11)
	at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:47)
	... 4 more
Caused by: java.lang.RuntimeException: Mixin transformation of net.minecraft.class_3532 failed
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:427)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.tryLoadClass(KnotClassDelegate.java:323)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:218)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:112)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	at net.minecraft.class_156.method_28122(class_156.java:138)
	at net.minecraft.class_156.<clinit>(class_156.java:85)
	... 7 more
Caused by: org.spongepowered.asm.mixin.transformer.throwables.MixinTransformerError: An unexpected critical error was encountered
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:392)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClass(MixinTransformer.java:234)
	at org.spongepowered.asm.mixin.transformer.MixinTransformer.transformClassBytes(MixinTransformer.java:202)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.getPostMixinClassByteArray(KnotClassDelegate.java:422)
	... 13 more
Caused by: java.lang.ClassCastException: class com.llamalad7.mixinextras.lib.apache.commons.tuple.ImmutablePair cannot be cast to class org.apache.commons.lang3.tuple.Pair (com.llamalad7.mixinextras.lib.apache.commons.tuple.ImmutablePair and org.apache.commons.lang3.tuple.Pair are in unnamed module of loader net.fabricmc.loader.impl.launch.knot.KnotClassLoader @38364841)
	at ca.fxco.memoryleakfix.config.mixinExtension.UnMixinExtension.preApply(UnMixinExtension.java:23)
	at org.spongepowered.asm.mixin.transformer.ext.Extensions.preApply(Extensions.java:156)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.preApply(TargetClassContext.java:413)
	at org.spongepowered.asm.mixin.transformer.TargetClassContext.applyMixins(TargetClassContext.java:401)
	at org.spongepowered.asm.mixin.transformer.MixinProcessor.applyMixins(MixinProcessor.java:363)
	... 16 more
```

</details>

**Comment by RaphiMC** [comment]
Removed memoryleakfix. Its known to cause issues with sodium

**Comment by Sunconure11** [comment]
I'm encountering the same issue, but on 1.19.2.

**Comment by Sunconure11** [comment]
https://pastebin.com/vzxC3xTN

**Comment by RaphiMC** [comment]
This is caused by memoryleakfix. Uninstall the mod or report the issue there

**Comment by RaphiMC** [comment]
MemoryLeakFix updated to v1.1.2 which fixes the issue

**Comment by Sunconure11** [comment]
Builds of Memory Leak Fix need to be uploaded for builds of 1.18.* and beyond to fully solve this

**Comment by RaphiMC** [comment]
Didn't notice that those versions aren't out yet
